miio/tests/test_protocol.py

```diff
@@ -32,3 +34,54 @@ def test_invalid_token(self):
             Utils.encrypt(payload, wrong_length)
         with self.assertRaises(ValueError):
             Utils.decrypt(payload, wrong_length)
+
+    def test_decode_json_payload(self):
+        token = bytes.fromhex(32 * '0')
+        ctx = {'token': token}
+
+        def build_msg(data):
+            encrypted_data = Utils.encrypt(data, token)
+
+            # header
+            magic = binascii.unhexlify(b'2131')
+            length = (32 + len(encrypted_data)).to_bytes(2, byteorder='big')
+            unknown = binascii.unhexlify(b'00000000')
+            did = binascii.unhexlify(b'01234567')
+            epoch = binascii.unhexlify(b'00000000')
+
+            checksum = Utils.md5(magic+length+unknown+did+epoch+token+encrypted_data)
+
+            return magic+length+unknown+did+epoch+checksum+encrypted_data
```

Comment:
Had to write my own message building function because Message.build() will always produce valid json but we want to explicitly test if Message.parse() can handle invalid json input.

Reply:
It's great to have tests for such behaviours, so thanks for this!